src/store/reducers/dropdown.ts

```diff
@@ -0,0 +1,44 @@
+import { Reducer } from "redux";
+
+import { List } from "immutable";
+import { DropdownAction, DropdownActionsType } from "../actions/dropdown";
+
+export enum DropdownType {
+  INFO = "info",
+  WARN = "warn",
+  ERROR = "error",
+  SUCCESS = "success"
+  // CUSTOM is not currently supported.
+}
+
+export interface DropdownMessage {
+  type: DropdownType;
+  id: string; // autogenerated
+  title: string;
+  message: string;
+}
+
+export type DropdownState = {
+  messages: List<DropdownMessage>;
```

Comment:
do we never clear this? should we remove message after we display it?

Reply:
message gets cleared via the `DISMISS_DROPDOWN_MESSAGE` action which gets fired by the `DropdownWrapper` when the dropdown message is closed.